### DISCUSSION TITLE
Verticals - Be able to append "Link" with the search box search string

### DISCUSSION THREAD

**Discussion by mikezimm** [microsoft-search/pnp-modern-search#1718] [Ideas]
This is a feature request regarding Verticals.

To avoid having to re-invent a search page from another source, it would be helpful in the Verticals to be able to give the link  URL like you have in the webpart to your documentation BUT....

Just append the link with whatever text was in the search box.

So like in your example you have a documentation vertical going to your documents:
https://microsoft-search.github.io/pnp-modern-search/

Say I have an enterprise system set up so that if I pass a search parameter in the Url, it goes to the page and does the text search from your search box.

So say in the Link URL property, I put this url (with the partial search paramter at the end):
https://myEnterpriseApplication.searchpage?searchword=

And say I type "findme" in the Pnp search box and then press my custom vertical....
It would up open this link in a new tab (just taking my Link URL and appending the text from the search box:
https://myEnterpriseApplication.searchpage?searchword=findme

I'm not thinking it would be hard but probably require some kind of connection between the Search Box and the Verticals webparts that is not currently there.

In the UI picture shown below, maybe you would have an Open Behaviour option such as "Append and Open in a new tab"

![image](https://user-images.githubusercontent.com/49648086/149411677-f415f8ca-b354-4674-996d-eeb241ca8a32.png)

**Comment by mikezimm** [comment]
@Laetterman ,
I did a quick search on this feature request and it sounds similar to this commit you made.
https://github.com/microsoft-search/pnp-modern-search/commit/d6ec61bac931ecf89799a5f8e305533254a6f8e7

I did some testing based on the code in the commit and I think it does address it by just inserting {SearchBoxQuery} where you want in your search....

So for instance:  https://www.google.com?q={SearchBoxQuery} does open a browser with google and that paramter.

One thing I did notice is that even though it opens in a new browser with the correct url, I still have to hit "Enter" to load that page with the search parameter.  Not sure why it doesn't load automatically.

## Small update after more testing...
If I type in a search string and then press the vertical without first hitting the search arrow or enter, it seems to pass a cached parameter, not necessarily what I just typed in.

**Comment by Laetterman** [comment]
Hi @mikezimm 

My fix was only that you can use variables {searchText} and Search box query text as variables in the verticals webpart Link URL. Is has nothing to do how the search result parg reacts on submittesd queries. This is part of the result page, how the evaluate and react on URL Parameters.

Therefore, it is a matter of the search environment (e.g. google) how they handle URL parameters and not the calling webpart. For Google you have to specify the sub-urtl /search that the query will automatically executed:

https://www.google.com/**search**?q=Marc

To have a result result page using modenr search webparts react on queries submitted by the URL you have to configure the search box webpart to accept dyxnamic datasources and also the search results webpart to accept dynamic values (sorry, only have german screenshots :-))

![image](https://user-images.githubusercontent.com/38464361/149466206-732b82dd-8746-4b40-a6ec-81fb1fd3b758.png)
![image](https://user-images.githubusercontent.com/38464361/149466360-4e453ad4-8a81-4d14-b348-4d4850ea9f23.png)
